Add php @global docs

shail-mehta opened the pull request:
Added `@global` documentation according to [PHP Doc Standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/) in below files:

1.  packages/block-library/src/query-no-results/index.php
2.  packages/block-library/src/query-pagination-next/index.php
3.  packages/block-library/src/query-pagination-numbers/index.php
4.  packages/block-library/src/template-part/index.php
5.  packages/widgets/src/blocks/widget-group/index.php


**Fixes:** https://github.com/WordPress/gutenberg/issues/59930

shail-mehta committed:
Added php @global docs

t-hamano commented on packages/block-library/src/query-no-results/index.php:
```diff
@@ -10,6 +10,8 @@
  *
  * @since 6.0.0
  *
+ * @global WP_Query $wp_query              WordPress Query object.
```
```suggestion
 * @global WP_Query $wp_query WordPress Query object.
```

If there are multiple global annotations, you need to insert a space according to the longest annotation. However, here there is only one global annotation, so a single space would be fine.

t-hamano commented on packages/block-library/src/query-pagination-next/index.php:
```diff
@@ -10,6 +10,8 @@
  *
  * @since 5.8.0
  *
+ * @global WP_Query $wp_query              WordPress Query object.
```
```suggestion
 * @global WP_Query $wp_query WordPress Query object.
```

t-hamano commented on packages/block-library/src/query-pagination-numbers/index.php:
```diff
@@ -10,6 +10,8 @@
  *
  * @since 5.8.0
  *
+ * @global WP_Query $wp_query              WordPress Query object.
```
```suggestion
 * @global WP_Query $wp_query WordPress Query object.
```

t-hamano commented on packages/block-library/src/template-part/index.php:
```diff
@@ -10,6 +10,8 @@
  *
  * @since 5.9.0
  *
+ * @global WP_Embed $wp_embed
```
```suggestion
 * @global WP_Embed $wp_embed WordPress Embed object.
```

t-hamano commented on packages/widgets/src/blocks/widget-group/index.php:
```diff
@@ -8,6 +8,9 @@
 /**
  * Renders the 'core/widget-group' block.
  *
+ * @global array $wp_registered_sidebars
+ * @global array $_sidebar_being_rendered
```
```suggestion
 * @global array      $wp_registered_sidebars
 * @global int|string $_sidebar_being_rendered
```

The type of `$_sidebar_being_rendered` should be int or string. This is because the argument value received [here](https://github.com/WordPress/gutenberg/blob/4bb17c47afb4741ebcfb54b1e498f27ed9831159/packages/widgets/src/blocks/widget-group/index.php#L62-L65) is assigned as is.

t-hamano commented on packages/widgets/src/blocks/widget-group/index.php:
```diff
@@ -59,6 +62,8 @@ function register_block_core_widget_group() {
  * it. This lets us get to the current sidebar in
  * render_block_core_widget_group().
  *
+ * @global array $_sidebar_being_rendered
```
```suggestion
 * @global int|string $_sidebar_being_rendered
```

t-hamano commented on packages/widgets/src/blocks/widget-group/index.php:
```diff
@@ -70,6 +75,8 @@ function note_sidebar_being_rendered( $index ) {
  * Clear whatever we set in note_sidebar_being_rendered() after WordPress
  * finishes rendering a sidebar.
+ *
+ * @global array $_sidebar_being_rendered
```
```suggestion
 * @global int|string $_sidebar_being_rendered
```

shail-mehta replied:
@t-hamano Thanks for the feedback.
 I have Updated all the changes as per suggested.

t-hamano replied:
Thanks for the update!

Finally, could you just commit to [this comment suggestion](https://github.com/WordPress/gutenberg/pull/59931/files#r1531458457)? I think everything else is fine 👍

shail-mehta replied:
Updated this change in latest commit.

shail-mehta committed:
Added php @global docs

t-hamano commented:
@shail-mehta Thanks for the update! Could you please rebase this PR once? This should eliminate any unit test failures.

shail-mehta committed:
Added php @global docs

shail-mehta committed:
Added php @global docs

shail-mehta committed:
Resolved Spacing Issue After @global doc

shail-mehta commented:
> @shail-mehta Thanks for the update! Could you please rebase this PR once? This should eliminate any unit test failures.

@t-hamano I have Updated  Remaining Changes as per suggested & also rebased this PR.

t-hamano reviewed (approved):
LGTM 👍
